templates/sprinklr-app/components/todo/TodoInput.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  useState,
+  useCallback,
+  KeyboardEventHandler,
+  KeyboardEvent,
+  memo,
+  ChangeEvent,
+  ChangeEventHandler,
+  useEffect,
+} from 'react';
+
+import { Textarea } from '@sprinklrjs/spaceweb/textarea';
+
+import { TextareaProps } from '@sprinklrjs/spaceweb/textarea/types';
+
+export type Props = {
+  text?: string;
+  placeholder?: string;
+  shouldSaveOnBlur?: boolean;
+  onSave: (text: string | undefined) => void;
+  className?: TextareaProps['className'];
+  overrides?: TextareaProps['overrides'];
+};
+
+const TodoInput = ({ text: _text, shouldSaveOnBlur, placeholder, onSave: _onSave, className, overrides }: Props) => {
+  const [text, setText] = useState<string | undefined>(_text);
+
+  useEffect(() => {
```

Comment:
change as discussed. create a ref and assign it to input and then we can use that everywhere

Reply:
stateful textarea not working as discussed